libminifi/src/utils/FlowFileQueue.cpp

```diff
@@ -30,70 +30,41 @@ bool FlowFileQueue::FlowFilePenaltyExpirationComparator::operator()(const value_
 }
 
 FlowFileQueue::value_type FlowFileQueue::pop() {
-  if (existsFlowFileWithExpiredPenalty()) {
-    value_type next_flow_file = priority_queue_.top();
-    priority_queue_.pop();
-    return next_flow_file;
+  if (empty()) {
+    throw std::logic_error{"pop() called on an empty FlowFileQueue"};
   }
 
-  if (!fifo_queue_.empty()) {
-    value_type next_flow_file = fifo_queue_.front();
-    fifo_queue_.pop();
-    return next_flow_file;
-  }
-
-  throw std::logic_error{"pop() called on FlowFileQueue when canBePopped() is false"};
-}
-
-/**
- * Pops any flow file off the queue, whether it has an unexpired penalty or not.
- */
-FlowFileQueue::value_type FlowFileQueue::forcePop() {
-  if (!fifo_queue_.empty()) {
-    value_type next_flow_file = fifo_queue_.front();
-    fifo_queue_.pop();
-    return next_flow_file;
-  }
-
-  if (!priority_queue_.empty()) {
-    value_type next_flow_file = priority_queue_.top();
-    priority_queue_.pop();
-    return next_flow_file;
-  }
-
-  throw std::logic_error{"forcePop() called on an empty FlowFileQueue"};
+  value_type next_flow_file = queue_.top();
```

Comment:
should we move from `queue_.top()`?

Reply:
I don't think we can: `top()` returns a const reference (and is a const member function).

Reply:
that is a bummer, there really is no non-const overload :(